### PR TITLE
[MAINT] lint & format repo; update CI; handle DeprWarns, etc.

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* lints and formats the `mapclassify` repo
  * several variables renamed due to `flake8` complaining
  * several variables/code chunks removed/commented out due to being flagged by `flake8` as not used
* add a pre-commit hook for formatting & listing
* drop support/testing for Python 3.7
* add linting to CI
* cleans up testing for warnings
* rerun notebooks
* cleans up and add doctests to CI run, and adapts for windows
* add random seed keyword args to `classifiers.MaxP`

***Still needs attention:*** (probably handle in a separate PR)
```
mapclassify/tests/test_mapclassify.py::TestUserDefined::test_UserDefined_invariant
  /Users/user/mapclassify/mapclassify/classifiers.py:908: RuntimeWarning: invalid value encountered in double_scalars
    gadf = 1 - self.adcm / Adam
```

xref: geopandas/geopandas#2553

### CODE REVIEW

**Review by martinfleis — COMMENTED** [review]
looks mostly okay to me, thanks!

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[run]
+include=mapclassify/*
+disable_warnings=include-ignored
+[report]
+omit = 
+    */datasets/*
+    docs/conf.py
+    *__init__.py
```
[the remark this answers is not in the document]
Comment (jGaboardi) [reply .coveragerc:8]: Fair point.

**Comment by jGaboardi** [comment]
> mostly okay

"mostly okay" is not a ringing endorsement. LOL. Anything specific to change/not change?

**Review by martinfleis — APPROVED** [review]
"mostly okay" referred to that one comment I left in the code :).

**Pull request merged**